jmetal-core/src/main/java/org/uma/jmetal/util/experiment/component/EvolutionChart/HistoryData.java

```diff
@@ -0,0 +1,162 @@
+            }
+            ps.print("\n");
+            for (int test = 0; test < numberOfTest; test++) {
+                for (int data = 0; data < numberOfData; data++) {
+                    ps.print(history[data][test]);
+                    if(data<numberOfData-1){
+                        ps.print("\t");
+                    }
+                }
+                ps.print("\n");
+            }
+        } catch (IOException ex) {
+            Logger.getLogger(GenerateEvolutionChart.class.getName()).log(Level.SEVERE, null, ex);
+        }
+    }
+    
+    
+    public static Map<String, Double> calculateQualityIndicator(List<DoubleSolution> population,
+                                                                String problemName,
+                                                                Set<String> indicatorToSave) {
+        Front referenceFront;
+        Map<String, Double> indicators = new HashMap();
+        try {
+            referenceFront = new ArrayFront("/resources/pareto_fronts/" + problemName + ".pf");
+            FrontNormalizer frontNormalizer = new FrontNormalizer(referenceFront);
+            Front normalizedReferenceFront = frontNormalizer.normalize(referenceFront);
+            Front normalizedFront = frontNormalizer.normalize(new ArrayFront(population));
+            List<PointSolution> normalizedPopulation = FrontUtils.convertFrontToSolutionList(normalizedFront);
+            //List<String> ind = new ArrayList(indicatorToSave);
+            for (String string : indicatorToSave) {
+                if(string.equals("HV")){
+                    indicators.put("HV", new PISAHypervolume<PointSolution>(normalizedReferenceFront).evaluate(normalizedPopulation));
+                } else if(string.equals("Epsilon")){
+                    indicators.put("Epsilon", new Epsilon<PointSolution>(normalizedReferenceFront).evaluate(normalizedPopulation));
+                } else if(string.equals("IGD")){
+                    indicators.put("IGD", new InvertedGenerationalDistancePlus<PointSolution>(normalizedReferenceFront).evaluate(normalizedPopulation));
+                } else if(string.equals("Spread")){
+                    indicators.put("Spread", new Spread<PointSolution>(normalizedReferenceFront).evaluate(normalizedPopulation));
+                }
+            }
+            
+        } catch (FileNotFoundException ex) { 
+            Logger.getLogger(GenerateEvolutionChart.class.getName()).log(Level.SEVERE, null, ex);
+        }
+        return indicators;
+    }
+    
+    
+    
+    public static void printAllDataInstances(Experiment<?,?> experiment, List<String> indicators){
+        System.out.println("Generating data files of algorithms evolution...");
+        for (ExperimentAlgorithm<?, ?> experimentAlgorithm : experiment.getAlgorithmList()) {  
+            for (String indicator : indicators) {
+                HistoryData hd = ((HistoricAlgorithm) experimentAlgorithm.getAlgorithm()).getHistory(indicator);
```

Comment:
Rather than taking the whole experiment, you should take only the relevant data, like `Collection<Map<Indicator, HistoryData>>`. If you need other stuff stored in the experiment structure, you should just ignore the structure and request what you need in argument. You don't need the whole experiment, only some relevant pieces, so don't depend on the whole stuff. We are in a single experiment tool, here, its scope should not be as broad as a whole experiment. `Experiment` should not be a God object, as explained in #221.